
## QM problem: Hamiltonian, eigenvalues and eigenvectors

$$\newcommand{\ket}[1]{\left|{#1}\right\rangle}$$
$$\newcommand{\bra}[1]{\left\langle{#1}\right|}$$
$$\newcommand{\braket}[2]{\left\langle{#1}\middle|{#2}\right\rangle}$$

Consider a quantum system in a state, $\ket{\Psi}$:

$\Psi = \frac{1}{\sqrt{3}}\left[\begin{array}{@{}c@{}}
    i \\
    -i \\
    i 
    \end{array} \right]$
    
The Hamiltonian is represented by the matrix shown below:


$\hat{H} = \left[\begin{array}{@{ }c@{\kern2em} c@{\kern2em} c@{ }}
2 & 1 & 0\\
1 & 2 & 0\\
0 & 0 & 3
\end{array} \right]$

(a) Determine the eigenvalues and eigenvectors of $\hat{H}$. What do the eigenvalues represent?


(b) Which eigenvalue of $\hat{H}$ is most likely to emerge from a measurement?


(c) Find $\langle H \rangle$, $\langle H^2 \rangle$, and $\sigma_H$.


### Solution:

(a) Determine the eigenvalues and eigenvectors of $\hat{H}$. What do the eigenvalues represent?

$\hat{H} = \left[\begin{array}{@{ }c@{\kern2em} c@{\kern2em} c@{ }}
2 & 1 & 0\\
1 & 2 & 0\\
0 & 0 & 3
\end{array} \right]$

In [1]:
# Import our usual libraries
import numpy as np

In [6]:
# Define the matrix

h_matrix = np.array([[2., 1., 0.]\
                   ,[1., 2., 0.]\
                   ,[0., 0., 3.]])

print("Hamiltonian is:\n", h_matrix)


Hamiltonian is:
 [[2. 1. 0.]
 [1. 2. 0.]
 [0. 0. 3.]]


In [7]:
# Let's use the eig() function

w_eigenvalues, v_eigenvectors = np.linalg.eig(h_matrix)

print("Eigenvalues are:\n", w_eigenvalues)

print("Eigenvectors are:\n", v_eigenvectors)

Eigenvalues are:
 [3. 1. 3.]
Eigenvectors are:
 [[ 0.70710678 -0.70710678  0.        ]
 [ 0.70710678  0.70710678  0.        ]
 [ 0.          0.          1.        ]]


(b) Which eigenvalue of $\hat{H}$ is most likely to emerge from a measurement?



$\Psi = \frac{1}{\sqrt{3}}\left[\begin{array}{@{}c@{}}
    i \\
    -i \\
    i 
    \end{array} \right]$

In [9]:
# Define the state vector

psi = np.array([1.j/np.sqrt(3), -1.j/np.sqrt(3), 1.j/np.sqrt(3)])

print(psi)

[ 0.+0.57735027j -0.-0.57735027j  0.+0.57735027j]


### Eigenvectors

In [11]:
vec_1 = v_eigenvectors[0]
vec_2 = v_eigenvectors[1]
vec_3 = v_eigenvectors[2]

print(vec_1)
print(vec_2)
print(vec_3)

[ 0.70710678 -0.70710678  0.        ]
[0.70710678 0.70710678 0.        ]
[0. 0. 1.]


$\bra{1}\cdot\ket{\psi}$

In [25]:
# np.inner()

p1 = np.dot(vec_1, psi)**2/np.dot(psi,psi)
p2 = np.dot(vec_2, psi)**2/np.dot(psi,psi)
p3 = np.dot(vec_3, psi)**2/np.dot(psi,psi)

print(np.abs(p1), np.abs(p2), np.abs(p3))

0.6666666666666665 0.0 0.33333333333333337


The most likely is $E_1 = 3$ for $\ket{\psi_1}=[ 0.70710678, -0.70710678 , 0.        ]$


(c) Find $\langle H \rangle$, $\langle H^2 \rangle$, and $\sigma_H$.


$\langle H \rangle$ = $\langle \psi | H | \psi \rangle$

In [27]:
H_exp = np.dot(np.dot(np.conjugate(psi),h_matrix), psi.T)

print(np.abs(H_exp))

1.6666666666666672


$\langle H^2 \rangle$ = $\langle \psi | H^2 | \psi \rangle$

In [28]:
H2_exp = np.dot(np.dot(np.conjugate(psi),h_matrix @ h_matrix)\
               , psi.T)

print(np.abs(H2_exp))

3.666666666666668


### Uncertianty:

$\sigma_H = \sqrt{\langle H^2 \rangle - \langle H\rangle^2}$

In [29]:
# Get sigma 
print("sigma = ", np.sqrt(np.abs(H2_exp) - np.abs(H_exp)**2))

sigma =  0.9428090415820631
